ouroboros-network/test/Test/Mux.hs

```diff
@@ -311,11 +431,16 @@ prop_mux_2_minis request0 response0 response1 request1 = ioProperty $ do
     clientAsync <- startMuxSTM client_mps client_w client_r sduLen Nothing
     serverAsync <- startMuxSTM server_mps server_w server_r sduLen Nothing
 
-    (\a b c d -> a .&&. b .&&. isNothing c .&&. isNothing d)
-      <$> verify_0
-      <*> verify_1
-      <*> wait clientAsync
-      <*> wait serverAsync
+
+    !r <- waitBoth clientAsync serverAsync
```

Comment:
It's likely that GHC does not need the bang pattern here, since you deconstruct the tuple in the following case expression. By inlining `waitBoth` GHC should be able to eliminate the tuple all together.

Reply:
Will remove the `!`.